internal/collector/postgres.go

```diff
@@ -272,3 +283,49 @@ func EnablePostgresLogging(
 		}
 	}
 }
+
+// generateLogFilenameAndRotationAge takes a retentionPeriod and returns a
+// log_filename and log_rotation_age to be used to configure postgres logging
+func generateLogFilenameAndRotationAge(
+	retentionPeriod *v1beta1.Duration,
+) (logFilename, logRotationAge string) {
+	number, interval := parseDurationForLogrotate(retentionPeriod)
+
+	// If the retention is a single hour or as single day, the log filename
+	// will not have any escape characters in it
+	logFilename = "postgresql.log"
+
+	// If the duration is 1 hour, set log_rotation_age to "1h"
+	if interval == "hourly" {
+		if number == 1 {
+			logRotationAge = "1h"
+		} else {
+			// If interval is "hourly" but duration is not 1 hour, we will round
+			// retention up to a day
+			logRotationAge = "1d"
+		}
+		// If the duration is not 1 hour (which is the minimum so it must be greater),
+		// then we will always set the rotation age to "1d"
+	} else {
+		logRotationAge = "1d"
+
+		// If the duration is greater than a day and less than or equal to a week,
+		// we will round retention up to a week
+		if number > 1 && number <= 7 {
+			logFilename = "postgresql-%w.log"
+		}
+
+		// If the duration is greater than a week and less than or equal to 28 days,
+		// we will round retention up to a month
+		if number > 7 && number <= 28 {
```

Comment:
What would it look like if we went up to 31? I would think, if a month had less than 31 days, PG wouldn't write over last month's last few days until the next month that had 31 days. That is, if Jan writes 1-31, then Feb writes 1-28, then 29, 30, 31 are all from Jan. Is that right?

Reply:
That's how I understand it.